api/app/Models/PoolCandidate.php

```diff
@@ -270,6 +270,20 @@ public function filterByEquity(Builder $query, array $equity): Builder
         return $query;
     }
 
+    public function filterByPoolCandidateStatus(Builder $query, ?string $poolCandidateStatus): Builder
```

Comment:
Oh! This is the perfect place for a [Laravel query scope](https://laravel.com/docs/9.x/eloquent#local-scopes)!